### ISSUE TITLE
Using AIMET compression schemes on Keras Model

### ISSUE THREAD

**Issue by navaravan** [quic/aimet#1742]
Hi @quic-ssiddego @quic-akhobare and other collaborators

> I have a Keras Training pipeline (which uses TF-2.4) with me, I want to know whether we can use AIMET compression schemes (AIMET compression API's) on it ? 

> From the below link I got to know its possible
https://forums.quicinc.com/t/compress-keras-tf-model/83

> I can understand from the AIMET compression API's is that it requires a session for doing all the operation, so how to extract the tf backend session from my keras model ?

**Comment by quic-akhobare** [comment]
Hi @navaravan - look at this API to get the backend TF session for a Keras model
https://www.tensorflow.org/api_docs/python/tf/compat/v1/keras/backend/get_session

However, one point to consider. For your case, you can use the TF session to compress the model using the AIMET TensorFlow model compression APIs. But for the subsequent fine-tuning (which is generally needed to recover accuracy post compression), you will need a training pipeline that works on a TF session. In other words, you need a TF training pipeline. Using a Keras training pipeline is not feasible.

AIMET model compression support for native Keras pipelines is still in the works and will be released sometime in the future. I cannot project a date at present.

**Comment by quic-akhobare** [comment]
If the above answers your question, please close the ticket.

**Comment by navaravan** [comment]
@quic-akhobare Thanks for your quick response
